### DISCUSSION TITLE
USB attached endoscope cam (with motion detection:)

### DISCUSSION THREAD

**Discussion by sudo-joe** [blakeblackshear/frigate#3776] [General]
Hello forum

would like to use/attach an USB endoscope camera which does work out of the box on Linuxes, like Mint.
Motion detection is required (no not for medical use:)

How are the chances that such a USB cam will work?

Thank's a lot!

Joe

**Comment by NickM-27** [comment]
If it works with ffmpeg then it'll work with frigate. We've had users use USB cams in the past.

**Comment by sudo-joe** [comment]
Thank's a lot NickM-27 for the promt feedback! What's the best approach to test that (if if do not yet own a running Frigate environment)?

Joe

**Reply by NickM-27** [reply]
Use ffmpeg to output the cameras feed to a mp4 file. Or just install frigate and look at what the other users who did this used.

**Comment by sudo-joe** [comment]
ffmpeg is not that easy...  Not possible with apps like VLC?

**Reply by NickM-27** [reply]
Frigate uses ffmpeg. I don't see how using the USB cam in VLC would help in the context of setting up frigate

**Comment by sudo-joe** [comment]
just wanna test the camera... Maybe it would be a good idea NickM-27,  to read my questions more carefully before posting :)

**Reply by NickM-27** [reply]
You are asking this question on the frigate repo so the assumption is when you ask 

> would like to use/attach an USB endoscope camera which does work out of the box on Linuxes

you are talking about using it in frigate.

If you are just asking generally how to test if the USB cam works with no intention of using it in frigate then that is off topic.

**Reply by NickM-27** [reply]
Also, VLC and ffmpeg are entirely different software. If you want to know if the USB cam will work with frigate then you need to use it with ffmpeg. If it works in VLC that in no way means it will work with ffmpeg / frigate?

**Comment by sudo-joe** [comment]
I do not yet own appropriate hw for a Frigate setup :(   Is there somebody who can tell me the right ffmpeg command? 
Thank you very much!


lsusb:
Bus 002 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 006 Device 001: ID 1d6b:0001 Linux Foundation 1.1 root hub
Bus 005 Device 002: ID 046d:c018 Logitech, Inc. Optical Wheel Mouse
Bus 005 Device 001: ID 1d6b:0001 Linux Foundation 1.1 root hub
Bus 001 Device 002: ID 1902:0327 Generic HD camera
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 004 Device 001: ID 1d6b:0001 Linux Foundation 1.1 root hub
Bus 003 Device 001: ID 1d6b:0001 Linux Foundation 1.1 root hub
..
..
dmesg:
[ 28.596742] IPv6: ADDRCONF(NETDEV_CHANGE): enp0s25: link becomes ready
161.427683] usb 1-4: new high-speed USB device number 2 using ehci-pci
[ 161.457111] usb 1-4: New USB device found, idVendor=1902, idProduct=0327, bcdDevice= 3.01
[ 161.457115] usb 1-4: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[ 161.457117] usb 1-4: Product: HD camera
[ 161.457120] usb 1-4: Manufacturer: Generic
[ 161.457122] usb 1-4: SerialNumber: 20201212000000
[ 161.499111] mc: Linux media interface: v0.10
[ 161.529235] videodev: Linux video capture interface: v2.00
[ 161.567247] uvcvideo: Found UVC 1.00 device HD camera (1902:0327)
[ 161.569384] uvcvideo 1-4:1.0: Entity type for entity Processing 2 was not initialized!
[ 161.569386] uvcvideo 1-4:1.0: Entity type for entity Camera 1 was not initialized!
[ 161.569638] input: HD camera : HD camera as /devices/pci0000:00/0000:00:1a.7/usb1/1-4/1-4:1.0/input/input13
[ 161.571339] usbcore: registered new interface driver uvcvideo
[ 161.571340] USB Video Class driver (1.1.1)

**Reply by NickM-27** [reply]
Check out https://github.com/blakeblackshear/frigate/issues/3213

It would be something like

`ffmpeg -f v4l2 -i /Dec/video -vf transpose=2,transpose=2 -c:v libx264 -preset ultrafast -tune zerolatency -b:v 600k -an output.mp4`

where /Dec/video is the by-id path of the USB cam.

May also need some changing based on your particular camera

**Comment by sudo-joe** [comment]
Thank's for the command! 

I hopend that the id path of the USB cam can be found in dmesg, but this does not seem to be case ;(

[dmesg-output:]
-> 1847.167315] input: HD camera : HD camera  as /devices/pci0000:00/0000:00:1a.7/usb1/1-3/1-3:1.0/input/input13

[Command:]
mint@mint:~$ ffmpeg -f v4l2 -i /devices/pci0000:00/0000:00:1a.7/usb1/1-3/1-3:1.0/input/input13 -vf transpose=2,transpose=2 -c:v libx264 -preset ultrafast -tune zerolatency -b:v 600k -an output.mp4

[Error-message]


[Error message:]
[video4linux2,v4l2 @ 0x564f67bcf740] Cannot open video device /devices/pci0000:00/0000:00:1a.7/usb1/1-3/1-3:1.0/input/input13: 
No such file or directory
/devices/pci0000:00/0000:00:1a.7/usb1/1-3/1-3:1.0/input/input13: No such file or directory


Question:
Where can the id path of the USB cam be found?

Thank's!
Joe

**Reply by NickM-27** [reply]
Check out https://askubuntu.com/a/848390

**Comment by sudo-joe** [comment]
mint@mint:/dev$ ls -ltrh /dev/video0
crw-rw----+ 1 root video 81, 0 sept. 19 17:45 /dev/video0

->ffmpeg -f v4l2 -i /dev/video0 -vf transpose=2,transpose=2 -c:v libx264 -preset ultrafast -tune zerolatency -b:v 600k -an output.mp

Thank you very much NickM for the help. The cam does work with ffmpeg :)
